> [ERROR] templates/: template: vcenter-operator/templates/secret.yaml:18:32: executing "vcenter-operator/templates/secret.yaml" at <include "resolve_secret">: error calling include: template: vcenter-operator/charts/utils/templates/_hosts.tpl:24:3: executing "resolve_secret" at <include "_resolve_secret" (tuple . false)>: error calling include: template: vcenter-operator/charts/utils/templates/_hosts.tpl:12:35: executing "_resolve_secret" at <$str>: invalid value; expected string

--- a/openstack/vcenter-operator/ci/test-values.yaml
+++ b/openstack/vcenter-operator/ci/test-values.yaml
@@ -6,3 +6,17 @@ imageVersion: ohr
 username: kopf
 password: arm
 tsig_key: bein
+role_id: fuss
+secret_id: zeh
+ad_ttu_username: hand
+ad_ttu_password: finger
+active_directory: hals
+vault_url: knie
+mount_point_read: rücken
+mount_point_write: bauch
+mount_point_write_nsxt: kinn
+max_time_not_seen: 0
+vault_check_interval: 1
+password_digits: 2
+password_length: 3
+password_symbols: 4

--- a/openstack/vcenter-operator/templates/secret.yaml
+++ b/openstack/vcenter-operator/templates/secret.yaml
@@ -6,8 +6,12 @@ metadata:
     {{- tuple . "vcenter-operator" "operator" | include "helm-toolkit.snippets.kubernetes_metadata_labels" | nindent 4 }}
 data:
   namespace: {{ .Values.global.keystoneNamespace | b64enc }}
+  {{- if .Values.username }}
   username: {{ .Values.username | include "resolve_secret" | b64enc }}
+  {{- end }}
+  {{- if .Values.global.vcenter_operator_master_password }}
   password: {{ .Values.global.vcenter_operator_master_password | include "resolve_secret" | b64enc }}
+  {{- end }}
   tsig_key: {{ .Values.tsig_key | include "resolve_secret" | b64enc }}
   manage_service_user_passwords: {{ .Values.manage_service_user_passwords | toString | b64enc }}
   {{- if .Values.manage_service_user_passwords }}

--- a/openstack/vcenter-operator/values.yaml
+++ b/openstack/vcenter-operator/values.yaml
@@ -25,4 +25,4 @@ owner-info:
     - Paul Pickhardt
   helm-chart-url: https://github.com/sapcc/helm-charts/tree/master/openstack/vcenter-operator
 
-manage_service_user_passwords: false
+manage_service_user_passwords: true

